thirdeye/thirdeye-frontend/app/pods/rootcause-aggregates-cache/service.js

```diff
@@ -56,12 +56,43 @@ export default Service.extend({
       return;
     }
 
-    // metrics
+    let metricUrnToOffestAndUrn = {};
     missing.forEach(urn => {
-      return this._fetchSlice(urn, requestContext);
+      const metricUrn = toMetricUrn(urn);
+      if (!(metricUrn in metricUrnToOffestAndUrn)) {
+        metricUrnToOffestAndUrn[metricUrn] = [];
+      }
+      metricUrnToOffestAndUrn[metricUrn].push([toAbsoluteUrn(urn, context.compareMode).split(':')[2].toLowerCase(), urn]);
     });
+
+    Object.keys(metricUrnToOffestAndUrn).forEach(
+        metricUrn => {
+          return this._fetchRowSlice(metricUrn, requestContext, metricUrnToOffestAndUrn);
+        });
+  },
+
+  _fetchRowSlice(metricUrn, context, metricUrnToOffestAndUrn){
+        const range = context.anomalyRange;
+        const offsets = metricUrnToOffestAndUrn[metricUrn].map(tuple => tuple[0]);
+        const urns = metricUrnToOffestAndUrn[metricUrn].map(tuple => tuple[1]);
+        const timezone = moment.tz.guess();
+        const url = `/rootcause/metric/aggregate/batch?urn=${metricUrn}&start=${range[0]}&end=${range[1]}&offsets=${offsets}&timezone=${timezone}`;
+        return fetch(url)
+        .then(checkStatus)
+        .then(res => this._extractAggregatesBatch(res, urns))
+        .then(res => this._complete(context, res))
+        .catch(error => this._handleError(urn, error));
+  },
+
+  _extractAggregatesBatch(incoming, urns) {
+    const aggregates = {};
+    for(var i = 0; i < urns.length; i++){
```

Comment:
nit: looks like a good candidate for a `reduce` function here

Reply:
documentation: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

Reply:
We are iterating through two arrays at the same time. Not sure how to do that with reduce in JS?

Reply:
ah, sorry didn't realize that. Nevermind then,  keep up the good work! 👍